GemCity.rb

```diff
@@ -1,56 +1,33 @@
 class GemCity
+  # This class represents the town of GemCity
+  # This is a town riddled with crime but we can find out how happy the town is
+  attr_reader :population
+  attr_accessor :officers, :thieves
 
-=begin
-This class represents the town of GemCity
-This is a town riddled with crime but we can find out how happy the town is
-=end
   def initialize
-    @people, @population = {
-     :thieves => 5, 
-       :Officers => 1},
-       50
-    end
-
-  def thieves thieves_number=@people[:thieves]
-     return @people[:thieves] = thieves_number
+    @thieves = 5
+    @officers = 1
+    @population = 50
   end
 
-  def officers
-    return @people[:Officers] 
+  def happiness_of_town
+    # happiness is random... people don't know what they want!
+    happiness = 0
+    population.times { happiness += rand((100 - successful_crime_rate)..100) }
+    happiness / 100
```

Comment:
100 is used throughout this file but to someone reading it might appear as a 'magic number'. You might want to use a constant to make some of these formulas more readable